azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement.go

```diff
@@ -0,0 +1,378 @@
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"display_name"},
+				RequiredWith:  []string{"origin_id"},
+				ValidateFunc:  validation.StringIsNotWhiteSpace,
+			},
+			"account_license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  licensing.AccountLicenseTypeValues.Express,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(licensing.AccountLicenseTypeValues.Advanced),
+					string(licensing.AccountLicenseTypeValues.EarlyAdopter),
+					string(licensing.AccountLicenseTypeValues.Express),
+					string(licensing.AccountLicenseTypeValues.None),
+					string(licensing.AccountLicenseTypeValues.Professional),
+					string(licensing.AccountLicenseTypeValues.Stakeholder),
+				}, true),
+				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
+					equalEntitlements := []string{
+						string(licensing.AccountLicenseTypeValues.EarlyAdopter),
+						string(licensing.AccountLicenseTypeValues.Express),
+						"basic",
+					}
+					stringInSlice := func(v string, valid []string) bool {
+						for _, str := range valid {
+							if strings.EqualFold(v, str) {
+								return true
+							}
+						}
+						return false
+					}
+					return strings.EqualFold(old, new) ||
+						(stringInSlice(old, equalEntitlements) && stringInSlice(new, equalEntitlements))
+				},
+			},
+			"licensing_source": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(licensing.LicensingSourceValues.Account),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(licensing.LicensingSourceValues.None),
+					string(licensing.LicensingSourceValues.Account),
+					string(licensing.LicensingSourceValues.Msdn),
+					string(licensing.LicensingSourceValues.Profile),
+					string(licensing.LicensingSourceValues.Auto),
+					string(licensing.LicensingSourceValues.Trial),
+				}, true),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+			"descriptor": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGroupEntitlementCreate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+	groupEntitlement, err := expandGroupEntitlement(d)
+	if err != nil {
+		return fmt.Errorf("Creating group entitlement: %v", err)
+	}
+
+	addedGroupEntitlement, err := addGroupEntitlement(clients, groupEntitlement)
+	if err != nil {
+		return fmt.Errorf("Creating group entitlement: %v", err)
+	}
+
+	flattenGroupEntitlement(d, addedGroupEntitlement)
+	return resourceGroupEntitlementRead(d, m)
+}
+
+func resourceGroupEntitlementRead(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+	groupEntitlementID := d.Id()
+	id, err := uuid.Parse(groupEntitlementID)
+	if err != nil {
+		return fmt.Errorf("Error parsing GroupEntitlementID: %s. %v", groupEntitlementID, err)
+	}
+	groupEntitlement, err := readGroupEntitlement(clients, &id)
```

Comment:
```suggestion
		groupEntitlement, err := clients.MemberEntitleManagementClient.GetGroupEntitlement(clients.Ctx, memberentitlementmanagement.GetGroupEntitlementArgs{
		GroupId: &id,
	})

```

Reply:
this is same as the function call, isn't it?

Reply:
Yes. The get operation only called once so there is no need to wrap it into a function

Reply:
Yeah, I realize that, all good, different coding style, no biggie, I've changed that.